mlflow/utils/openai_utils.py

```diff
@@ -75,23 +76,37 @@
 REQUEST_FIELDS = REQUEST_FIELDS_CHAT | REQUEST_FIELDS_COMPLETIONS | REQUEST_FIELDS_EMBEDDINGS
 
 
+def mock_raise_for_status(status_code):
```

Comment:
curious - trying to learn python
what is the benefit of moving this func out of the closure?

Reply:
It is just to avoid repetition. For more context, I first added this function for all 4 types of mock response (normal, async, stream, astream) so extracted out. Then it turns out I only need two - so the benefit becomes less but still no reason to revert:)